src/plugins/newsfeed/public/plugin.tsx

```diff
@@ -47,27 +48,33 @@ export class NewsfeedPublicPlugin implements Plugin<Setup, Start> {
     });
   }
 
-  public setup(core: CoreSetup): Setup {
+  public setup(core: CoreSetup) {
     return {};
   }
 
-  public start(core: CoreStart): Start {
-    const api$ = this.fetchNewsfeed(core);
+  public start(core: CoreStart) {
+    const api$ = this.fetchNewsfeed(core, this.config).pipe(share());
     core.chrome.navControls.registerRight({
       order: 1000,
       mount: (target) => this.mount(api$, target),
     });
 
-    return {};
+    return {
+      createNewsFeed$: (endpoint: NewsfeedApiEndpoint) =>
+        this.fetchNewsfeed(core, { service: { pathTemplate: `/${endpoint}/v{VERSION}.json` } }),
```

Comment:
we are dropping all the other expected config parameters in `NewsfeedPluginBrowserConfig` (that is why we have to add all the fallback calls).

I'm wondering if it would be better to pass the `pathTemplate` parameter to `fetchNewsfeed` and the function will merge it with the existing `this.config` before invoking `getApi`?

Reply:
Yeah I think that makes sense. I added support for the entire newsfeed config object in f42e962.